File: src/MaksIT.WebUI/src/models/certsUI/apiKeys/CreateApiKeyEntityScopeRequest.ts
import { RequestModelBase } from '../../RequestModelBase'
import { ScopeEntityType, ScopePermission } from '../../engine/scopeEnums'

/** Mirrors <c>MaksIT.CertsUI.Models.APIKeys.CreateApiKeyEntityScopeRequest</c>. */
export interface CreateApiKeyEntityScopeRequest extends RequestModelBase {
  entityId: string
  entityType: ScopeEntityType
  scope: ScopePermission
}

File: src/MaksIT.CertsUI/Models/APIKeys/CreateApiKeyEntityScopeRequest.cs
using MaksIT.Core.Abstractions.Webapi;
using MaksIT.CertsUI.Engine;

namespace MaksIT.CertsUI.Models.APIKeys;

public class CreateApiKeyEntityScopeRequest : RequestModelBase {
  /// <summary>
  /// The ID of the entity (e.g., organization or application).
  /// </summary>
  public Guid EntityId { get; set; }

  /// <summary>
  /// The type of the entity (e.g., ", "Application").
  /// </summary>
  public ScopeEntityType EntityType { get; set; }

  /// <summary>
  /// The scope granted to the user for this entity.
  /// </summary>
  public ScopePermission Scope { get; set; }
}